smart-web-server/src/main/java/org/smartdata/server/mappers/CmdletInfoMapper.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.smartdata.server.mappers;
+
+import org.mapstruct.Condition;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+import org.mapstruct.ReportingPolicy;
+import org.mapstruct.ValueMapping;
+import org.smartdata.metastore.model.SearchResult;
+import org.smartdata.model.CmdletInfo;
+import org.smartdata.model.CmdletState;
+import org.smartdata.model.TimeInterval;
+import org.smartdata.model.request.CmdletSearchRequest;
+import org.smartdata.server.generated.model.CmdletDto;
+import org.smartdata.server.generated.model.CmdletStateDto;
+import org.smartdata.server.generated.model.CmdletsDto;
+import org.smartdata.server.generated.model.StateChangeTimeIntervalDto;
+import org.smartdata.server.generated.model.SubmissionTimeIntervalDto;
+
+import java.util.List;
+
+@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.ERROR)
```

Comment:
Use MapStruct with caution. In other projects, we have encountered its non-obvious behavior at runtime.